packages/blockchain-extension/extension/explorer/gatewayExplorer.ts

```diff
@@ -166,67 +173,46 @@ export class BlockchainGatewayExplorerProvider implements BlockchainExplorerProv
     }
 
     private async createConnectionTree(): Promise<BlockchainTreeItem[]> {
+
         const tree: BlockchainTreeItem[] = [];
 
         const allGateways: FabricGatewayRegistryEntry[] = await this.fabricGatewayRegistry.getAll();
 
-        if (allGateways.length === 0) {
-            tree.push(new TextTreeItem(this, 'No gateways found'));
-        } else {
-            for (const gateway of allGateways) {
-
-                const command: vscode.Command = {
-                    command: ExtensionCommands.CONNECT_TO_GATEWAY,
-                    title: '',
-                    arguments: [gateway]
-                };
-
-                const gatewayName: string = gateway.displayName ? gateway.displayName : gateway.name;
-
-                let environmentEntry: FabricEnvironmentRegistryEntry;
-                let runtime: LocalEnvironment | ManagedAnsibleEnvironment;
-                if (gateway.fromEnvironment) {
-                    environmentEntry = await FabricEnvironmentRegistry.instance().get(gateway.fromEnvironment);
-
-                }
+        const gatewayGroups: Array<FabricGatewayRegistryEntry[]> = [];
+        for (const gateway of allGateways) {
+            if (gatewayGroups.length === 0) {
+                gatewayGroups.push([gateway]);
+                continue;
+            }
 
-                if (environmentEntry && environmentEntry.managedRuntime) {
+            if (gateway.fromEnvironment && gateway.fromEnvironment === gatewayGroups[gatewayGroups.length - 1][0].fromEnvironment) {
```

Comment:
this is abit icky,, i don't like that you are relying on them being in alphabetical order. Think there is probably a better way of writting it